src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/Interop/VsHierarchyExtensions.cs

```diff
@@ -54,5 +54,18 @@ public static T GetProperty<T>(this IVsHierarchy hierarchy, HierarchyId item, Vs
             // NOTE: We consider it a bug in the underlying project system or the caller if this cast fails
             return (T)resultObject;
         }
+                
+        /// <summary>
+        /// Returns EnvDTE.Project object for the hierarchy
+        /// </summary>
+        public static EnvDTE.Project GetDTEProject(this IVsHierarchy hierarchy)
+        {
```

Comment:
Can you replace usage of this method with the generic one above it? I'm not a fan of the overly defensive casts, and HRESULT checks - as they hide bugs. It would a bug if our hierarhcy didn't have a extensibility object that didn't implement EnvDTE.Project